c_glib/arrow-glib/reader.cpp

```diff
@@ -1497,6 +1501,393 @@ garrow_csv_reader_read(GArrowCSVReader *reader,
+                          static_cast<GParamFlags>(G_PARAM_READWRITE));
+  g_object_class_install_property(gobject_class,
+                                  PROP_JSON_READER_BLOCK_SIZE,
+                                  spec);
+
+
+  auto parse_options = arrow::json::ParseOptions::Defaults();
+
+  /**
+   * GArrowJSONReadOptions:allow-newlines-in-values:
+   *
+   * Whether objects may be printed across multiple lines (for example pretty printed).
+   * if %FALSE, input must end with an empty line.
+   *
+   * Since: 0.14.0
+   */
+  spec = g_param_spec_boolean("allow-newlines-in-values",
+                              "Allow newlines in values",
+                              "Whether objects may be printed across multiple lines "
+                              "(for example pretty printed). "
+                              "if %FALSE, input must end with an empty line.",
+                              parse_options.newlines_in_values,
+                              static_cast<GParamFlags>(G_PARAM_READWRITE));
+  g_object_class_install_property(gobject_class,
+                                  PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES,
+                                  spec);
+
+  /**
+   * GArrowJSONReadOptions:unexpected-field-behavior:
+   *
+   * How to parse handle fields outside the explicit schema.
+   *
+   * Since: 0.14.0
+   */
+  spec = g_param_spec_enum("unexpected-field-behavior",
+                           "UnexpectedFieldBehavior",
+                           "How to parse handle fields outside the explicit schema.",
+                           GARROW_TYPE_JSON_READ_UNEXPECTED_FIELD_BEHAVIOR,
+                           GARROW_JSON_READ_INFERTYPE,
+                           static_cast<GParamFlags>(G_PARAM_READWRITE));
+  g_object_class_install_property(gobject_class,
+                                  PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR,
+                                  spec);
+}
+
+/**
+ * garrow_json_read_options_new:
+ *
+ * Returns: A newly created #GArrowJSONReadOptions.
+ *
+ * Since: 0.14.0
+ */
+GArrowJSONReadOptions *
+garrow_json_read_options_new(void)
+{
+  auto json_read_options = g_object_new(GARROW_TYPE_JSON_READ_OPTIONS,
+                                        "pool", arrow::default_memory_pool(),
+                                        NULL);
+  return GARROW_JSON_READ_OPTIONS(json_read_options);
+}
+
+/**
+ * garrow_json_read_options_set_explicit_schema:
+ * @options: A #GArrowJSONReadOptions.
+ * @schema: The optional explicit #GArrowSchema.
+ *
+ * Set optional explicit schema for passing custom conversion rules.
+ *
+ * Since: 0.14.0
+ */
+void
+garrow_json_read_options_set_explicit_schema(GArrowJSONReadOptions *options,
+                                             GArrowSchema *schema)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(options);
+  auto arrow_schema = garrow_schema_get_raw(schema);
+  priv->parse_options.explicit_schema = arrow_schema;
+}
```

Comment:
Could you implement this as a property?
We can use `g_param_spec_object()`.

Reply:
We can use `schema` as property name. We can omit `explicit-` because we aren't using `explicit-` for `GArrowCSVReadOptions`.

Reply:
Yes. I've fixed them.